Use the OpenSSL 1.1 and later API: SSLv2_client_method() no longer exists
and CRYPTO_lock has been replaced by CRYPTO_free.

Backported from the Debian partimage package
(debian/patches/02-openssl.patch and debian/patches/03-openssl11.patch),
original authors include Chris West <git@goeswhere.com>, Debian bug
#859549. The configure.ac change is required because autoreconf is run
during the build.

diff --git a/configure.ac b/configure.ac
index bb2ff61..267a140 100644
--- a/configure.ac
+++ b/configure.ac
@@ -240,7 +240,7 @@ if test "$SSL" = "yes"; then
   AC_CHECKING([ for SSL Library and Header files ... ])
   AC_SEARCH_HEADERS(rsa.h crypto.h x509.h pem.h ssl.h err.h,
     $SSL_HDR_DIR /usr/include/ssl /usr/include/openssl /usr/include,
-    [  AC_CHECK_LIB(crypto, CRYPTO_lock, [LIBS="$LIBS -lcrypto"],
+    [  AC_CHECK_LIB(crypto, CRYPTO_free, [LIBS="$LIBS -lcrypto"],
         AC_MSG_ERROR([ Required for SSL Crypto Library not found. ])
       )
        AC_CHECK_LIB(ssl, SSL_CTX_new,
diff --git a/src/client/netclient.cpp b/src/client/netclient.cpp
index 30b8d5c..43b2672 100644
--- a/src/client/netclient.cpp
+++ b/src/client/netclient.cpp
@@ -43,9 +43,8 @@ CNetClient::CNetClient(bool bMustLogin, bool bUseSSL):CNet()
     {
       showDebug(3, "initializing client ssl\n");
       SSLeay_add_ssl_algorithms();
-      meth = SSLv2_client_method();
       SSL_load_error_strings();
-      ctx = SSL_CTX_new(meth);
+      ctx = SSL_CTX_new(SSLv23_client_method());
       if (!ctx)
         THROW(ERR_SSL_CTX);
       m_bUseSSL = (ctx != NULL);
diff --git a/src/client/netclient.h b/src/client/netclient.h
index 8423798..ea3b3ca 100644
--- a/src/client/netclient.h
+++ b/src/client/netclient.h
@@ -35,11 +35,6 @@ private:
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * server_cert;
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
-  SSL_METHOD const * meth;
-#else
-  SSL_METHOD * meth;
-#endif // OPENSSL_VERSION_NUMBER
 
 #endif
   bool m_bUseSSL;
diff --git a/src/server/netserver.cpp b/src/server/netserver.cpp
index b3ba1c7..0652d49 100644
--- a/src/server/netserver.cpp
+++ b/src/server/netserver.cpp
@@ -39,8 +39,7 @@ CNetServer::CNetServer(unsigned short int port):CNet()
     {
       SSL_load_error_strings();
       SSLeay_add_ssl_algorithms();
-      meth = SSLv23_server_method();
-      ctx = SSL_CTX_new(meth);
+      ctx = SSL_CTX_new(SSLv23_server_method());
       if (!ctx)
         {
           ERR_print_errors_fp(stderr);
diff --git a/src/server/netserver.h b/src/server/netserver.h
index fc057bc..fa2343f 100644
--- a/src/server/netserver.h
+++ b/src/server/netserver.h
@@ -41,11 +41,6 @@ private:
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * client_cert;
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
-  SSL_METHOD const * meth; 
-#else
-  SSL_METHOD * meth;
-#endif // OPENSSL_VERSION_NUMBER
  
   int err;
 #endif
